apps/speed.c

```diff
@@ -1040,33 +1038,31 @@ static int ECDSA_verify_loop(void *args)
 /* ******************************************************************** */
 static long ecdh_c[EC_NUM][1];
 
-static int ECDH_compute_key_loop(void *args)
+static void ECDH_EVP_derive_key(unsigned char *derived_secret,
+                            size_t *outlen,
+                            EVP_PKEY_CTX *ctx)
 {
-    loopargs_t *tempargs = *(loopargs_t **)args;
-    EC_KEY **ecdh_a = tempargs->ecdh_a;
-    EC_KEY **ecdh_b = tempargs->ecdh_b;
-    unsigned char *secret_a = tempargs->secret_a;
+    if( !EVP_PKEY_derive(ctx, derived_secret, outlen) ) {
+        /* FIXME: handle errors */
+        ;
```

Comment:
If you're going to handle errors, it would be a good idea if this function returned `int` rather than `void`...